com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -844,35 +975,39 @@ private void Awake()
             }
         }
 
-
         /// <inheritdoc/>
         public override void OnNetworkSpawn()
         {
-            // must set up m_Transform in OnNetworkSpawn because it's possible an object spawns but is disabled
-            //  and thus awake won't be called.
-            // TODO: investigate further on not sending data for something that is not enabled
-            m_Transform = transform;
-            m_ReplicatedNetworkState.OnValueChanged += OnNetworkStateChanged;
-
-            CanCommitToTransform = IsServer;
             m_CachedIsServer = IsServer;
             m_CachedNetworkManager = NetworkManager;
 
+            Initialize();
+
+            // This assures the initial spawning of the object synchronizes all connected clients
+            // with the current transform values. This should not be placed within Initialize since
+            // that can be invoked when ownership changes.
             if (CanCommitToTransform)
             {
-                TryCommitTransformToServer(m_Transform, m_CachedNetworkManager.LocalTime.Time);
-            }
-            m_LocalAuthoritativeNetworkState = m_ReplicatedNetworkState.Value;
+                // Teleport to current position
+                SetStateInternal(transform.position, transform.rotation, transform.localScale, true);
 
-            // crucial we do this to reset the interpolators so that recycled objects when using a pool will
-            //  not have leftover interpolator state from the previous object
-            Initialize();
+                // Force the state update to be sent
```

Comment:
I think it's interesting here that the comment says "Force" but the method says "Try". I'm sure it does what the comment says it does, but it's an interesting contradiction of language.

Reply:
Yeah... I think we still have some names to update in NetworkTransform.